### PR TITLE
[MODULAR] [READY] A new station map, Ringworm!

### PR DESCRIPTION
**Pull request by zydras**
# Ready to Testmerge!

## About The Pull Request

Wow hey it's been a while we had a new Skyrat map, hasn't it?
**Welcome to Ringworm!** A new Nanotrasen station located right in the middle of Lavaland's asteroid ring. With a lunar ambience, the station is surrounded with a great quantity of asteroids that are still rich with minerals for the crew to go harvest.
<details>
<summary>Full map</summary>

## Lower deck:

</details>

## How This Contributes To The Skyrat Roleplay Experience

Naturally, new Skyrat maps means we can experiment with more fun ideas that TG may not be compatible for (notably good places to put blueshield/NTrep offices), but it's also more diversity! As for the map itself,
- Asteroid mining has always been something that people have requested comes back. This station adds it in tandem with lavaland! It's a low risk, low reward environment with a few rich spots well guarded by Hivelords (The asteroid equivalent of lavaland legions)
- Wanting to make a station unique, I opted to go with a circular shape for the entire station's general shape. This way, not only is the station shape funky, it also means every department gets a healthy dose of exposure to the central hallway, including enginnering! (A rarity for maps, let's be real). It's like donutstation if it was good! 
- Another detail I really wanted to work on was proper use of 2Z in a way that felt dynamic, and not for specific use cases. This means each department has its own unique way of approaching multi-z details, wether it's security getting a panopticon for the warden, engineering having fun with a multi-z supermatter, cargo getting that good old rugged warehouse style or service having a lounge that overlooks the diner (and many little rooms being 2z, which includes many maintenance halls!) 
And a few more details worth playing the map for :3c

## Proof of Testing

<details>
<summary>A variety of screenshots from the station during local tests</summary>
  
Screenshot (2208)
Screenshot (2210)
Screenshot (2217)
Screenshot (2226)
Screenshot (2227)
Screenshot (2239)
Screenshot (2232)

</details>

## Changelog

:cl:
add: Adds an entire new lowpop station, Ringworm!
code: Added airless variants of moon turfs and created moon asteroid turfs
config: Ringworm added to rotation
/:cl:

### CODE REVIEW

**Comment by Stalkeros2** [comment]
ASTEROOOOOID WOOOOOOOOOOOOOOO

**Comment by lessthnthree** [comment]
Full map not revealing anything was the biggest moment of anticipation followed by letdown

**Comment by zydras** [comment]
> Full map not revealing anything was the biggest moment of anticipation followed by letdown

Admittedly I wanted to add the mapdiffbot's icon but. The asteroids were straight up blanked out from the images it produced so I can't use that one. Thanks for reminding me to use the SDMM screenshot though, it's my only alternative 😭

**Comment by W4rd3nn** [comment]
This is so hot wanna make out?

In all seriousness, this looks awesome. You've taken basically all the feedback we gave you and ran with it. Every department getting to play with 2Z is a really welcome addition. My only criticism being its population category. This doesn't feel like a lowpop map. Midpop, for sure. I think capping it at 66 will mean it barely ever gets played, which would be a huge slap in the face for all this work you've done.

I'd maybe go for an 80-100 cap, or just uncap it entirely. Birdshot has a 66 cap and this isn't really a birdshot sized map.

**Comment by zydras** [comment]
If you say so! I'll edit that.

**Comment by W4rd3nn** [comment]
Worst case scenario, we can always dial it back if it proves to be too cramped

**Comment by burgerenergy** [comment]
Just full cap and ask for an emoji vote if it needs to be reduced once play tested

go sicko

**Comment by zydras** [comment]
There's a few things that I would like to get help on (A few integrations tests) but otherwise the map is ready!

File: _maps/ringworm.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+	"version": 1,
+	"map_name": "ringworm",
```
Comment (Iajret) [inline _maps/ringworm.json:3]: you called it RingWorm in `area_spawn_subsystem` and that's why you failing area_spawn unit tests (well, that's not THE reason, but i think you placed all skyrat stuff yourself, right?)

Comment (zydras) [reply _maps/ringworm.json:3]: Yeah, I had issues with a previous linter due to the map folder being capitalized (for some god-forsaken reason??) and forgot to edit this one, thanks for the catch!

**Comment by Iajret** [comment]
as for other unit test for your map. They give you exact coords. Mapping verbs help finding this things but whatever.
<details>
<summary>Missing cable</summary>
  
image
</details>

<details>
<summary>Space turf.</summary>

image
</details>

I not a mapping guru myself... but i wonder if there's another way to give asteroids darkness without painting them with stupid `Station Asteroid` zones. As engie main first thing i would've wanted to try is to build... well, anything actually, inside this rocks. But you can't place APCs inside station asteroid. At least that was the case for Tram Station.

And, for the shuttle, well........

File: modular_skyrat/modules/advanced_shuttles/code/shuttles.dm [hunk 2]
```diff
@@ -128,6 +128,16 @@
 /datum/map_template/shuttle/cargo/skyrat/delta
 	suffix = "delta_skyrat"	//I hate this. Delta station is one tile different docking-wise, which fucks it ALL up unless we either a) change the map (this would be nonmodular and also press the engine against disposals) or b) this (actually easy, just dumb)
 
+/datum/map_template/shuttle/cargo/skyrat/delta
```
Comment (Iajret) [inline modular_skyrat/modules/advanced_shuttles/code/shuttles.dm]: you messed with pathes, probably? You redefined Delta's shuttle and that's why you getting errors for Deltastation and Raptor

Comment (zydras) [reply modular_skyrat/modules/advanced_shuttles/code/shuttles.dm]: my ass straight up forgot to change the path name 😭 thanks for pointing this out lmao

**Comment by Arctaisia** [comment]
Hell yeah, love it!!!!

**Comment by Elli-Skala** [comment]
You need to do a double look over the xenobio egg pen. The shield walls can be easily broken from the inside

**Comment by zydras** [comment]
Boy This Sure Is A Lot Of Merging From Master Haha!

**Comment by Jolly-66** [comment]
Drafting this for the time being until we (maptainers) can get around to putting notes together to review this.

Also I believe drafting PRs makes them not stale (don't quote me on it). If it doesn't, I'll slap the stale-b-gone label on it then.

**Comment by W4rd3nn** [comment]
i miss dis :(

**Comment by Jolly-66** [comment]
Alas, Zydras left the Discord and I do not have contact with them.
I do want to see this map come to fruition, however, its Zydras decision to be made if they wish to have the development continued.

I'm going to be closing this PR as I'm assuming Zydras has moved on, or is taking a break. They may reach out to me if they wish to have this reopened, or if someone else wishes to carry their torch forward, they may do so with their permission.

Thank you for your hard work Zydras.

**Comment by Arctaisia** [comment]
> Alas, Zydras left the Discord and I do not have contact with them.
> I do want to see this map come to fruition, however, its Zydras decision to be made if they wish to have the development continued.
> 
> I'm going to be closing this PR as I'm assuming Zydras has moved on, or is taking a break. They may reach out to me if they wish to have this reopened, or if someone else wishes to carry their torch forward, they may do so with their permission.
> 
> Thank you for your hard work Zydras.

I believe Zydras has indeed moved on! They left SS13 entirely to instead be within the Korps community, just as we have. We're only here because we forgot we subscribed and got a notification!

But aye, they're unlikely to return for the time being, even for a long time. What becoming a queer supervillain does to a girl tbh...

**Pull request closed without merging**